scripts/gen-auth-api-spec.ts

```diff
@@ -130,6 +128,70 @@ async function toOpenapi3(discovery: any): Promise<any> {
   return openapi3;
 }
 
+const pathParamsForFlatPathParam: Map<string, string> = new Map([
+  ["{projectsId}", "{targetProjectId}"],
+  ["{tenantsId}", "{tenantId}"],
+]);
+
+const paramPattern = /{([^}]+)}/g;
+
+function replaceWithFlatPath(discovery: { [key: string]: any }): void {
+  if (discovery.methods) {
+    Object.entries(discovery.methods).forEach(([_, method]) => {
+      if (typeof method === "object") {
+        // Replace flat path param names with path param names
+        // e.g. for endpoint identitytoolkit.projects.defaultSupportedIdpConfigs.get:
+        // path = v2/{name}
+        // flatPath = v2/projects/{projectsId}/defaultSupportedIdpConfigs/{defaultSupportedIdpConfigsId}
+        //
+        // transform v2/projects/{projectsId}/defaultSupportedIdpConfigs/{defaultSupportedIdpConfigsId}
+        //       --> v2/projects/{targetProjectId}/defaultSupportedIdpConfigs/{defaultSupportedIdpConfigsId}
+        let flatPath = (method as any).flatPath;
```

Comment:
Avoid `any` by casting it as:
`let flatPath = (method as {flatPath: string}).flatPath`